src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -235,40 +235,36 @@ private void setupKeyboardShortcuts() {
                 KeyPress.setLastKeyPressedCodeAndTime(event.getCode());
             }
             if (NEW_COMMENT.match(event)) {
-                if (KeyPress.isValidKeyCombination(GOTO_MODIFIER.getCode(), event.getCode())) {
-                    ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
-                } else {
+                TurboIssue selected = getCurrentlySelectedIssue();
+                if (KeyPress.isValidKeyCombination(GOTO_MODIFIER.getCode(), event.getCode())
+                    && selected.isPullRequest()) { // goto PR discussion tab
+
+                    ui.getBrowserComponent().openPrTab(selected, DISCUSSION_TAB);
+                } else { // focus new comment textbox
                     if (!ui.getBrowserComponent().isCurrentUrlIssue()) {
-                        openPageOfCurrentlySelectedIssue();
-                    } else if (!ui.getBrowserComponent().isCurrentUrlDiscussion()) {
-                        ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
+                        openIssuePage(selected);
+                    } else if (ui.getBrowserComponent().isCurrentUrlPR()) {
+                        ui.getBrowserComponent().openPrTab(getCurrentlySelectedIssue(), DISCUSSION_TAB);
```

Comment:
This whole thing can be changed. Originally, I need to check `isCurrentUrlDiscussion` in case I am in other tabs of PR page. And it was a quick fix, so I didn't bother refactoring.

What you can do here is replace the whole `if else` case to
```
if (!ui.getBrowserComponent().isCurrentUrlIssueDiscussion()) {
  openIssuePage(selected);
}
.
.
.
private boolean isCurrentUrlDiscussion() {
// Use GitHub URL check, but create a new regex where it only matches issues / pull url with no suffix of commit or files.
}
```
This is because `showIssue` in `openIssue` handles both PR and issue, and opening default PR page will direct you to the discussion page, anyway. This leads to cleaner `if` case.